bionic/deriver.py

```diff
@@ -468,7 +468,7 @@ def _process_entry(self, entry, task_key_logger):
             or self._bootstrap.process_executor is None
             # This is a non-serializable entity that needs to be returned.
             or (
-                not state.provider.attrs.should_persist()
+                not state.should_persist
```

Comment:
This is going to conflict with your PR. I'll rebase my PR on latest master once you merge your PR.